core/lib/prover_interface/src/api.rs

```diff
@@ -24,7 +24,7 @@ pub struct ProofGenerationDataRequest {}
 
 #[derive(Debug, Serialize, Deserialize)]
 pub enum ProofGenerationDataResponse {
-    Success(Option<ProofGenerationData>),
+    Success(Box<Option<ProofGenerationData>>),
```

Comment:
`Box<Option<_>>` looks weird; `Option<Box<_>>` is a more idiomatic alternative since it has the same memory layout as `Box<_>`.